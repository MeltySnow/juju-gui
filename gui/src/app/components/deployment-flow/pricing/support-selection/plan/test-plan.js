/* Copyright (C) 2018 Canonical Ltd. */
'use strict';

const React = require('react');
const enzyme = require('enzyme');

const Analytics = require('test/fake-analytics');
const DeploymentSupportSelectionPlan = require('./plan');

describe('DeploymentSupportSelectionPlan', () => {

  const renderComponent = (options = {}) => enzyme.shallow(
    <DeploymentSupportSelectionPlan
      analytics={Analytics}
      classes={options.classes}
      features={options.features || ['feature 1', 'feature 2']}
      hourPrice={options.hourPrice || '0.5'}
      machineCount={10}
      onSelect={options.onSelect || sinon.stub()}
      selected={options.selected === undefined ? false : options.selected}
      title={options.title || 'Gold'} />
  );

  it('can render', function() {
    const wrapper = renderComponent();
    expect(wrapper).toMatchSnapshot();
  });

  it('can display as selected', function() {
    const wrapper = renderComponent({selected: true});
    assert.equal(
      wrapper.prop('className').includes('deployment-support-select-plan--selected'),
      true);
  });

  it('can handle extra classes', function() {
    const wrapper = renderComponent({classes: ['extra']});
    assert.equal(wrapper.prop('className').includes('extra'), true);
  });
});
