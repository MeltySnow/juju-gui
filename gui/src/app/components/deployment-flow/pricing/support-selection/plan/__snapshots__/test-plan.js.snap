// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeploymentSupportSelectionPlan can render 1`] = `
<div
  className="deployment-support-select-plan four-col"
>
  <div
    className="deployment-support-select-plan__card"
    onClick={[Function]}
    role="button"
    tabIndex="0"
  >
    <h3>
      Gold
    </h3>
    <ul
      className="deployment-support-select-plan__features"
    >
      <li
        className="deployment-support-select-plan__feature"
        key="feature 10"
      >
        <SvgIcon
          name="bullet"
          size="14"
        />
        feature 1
      </li>
      <li
        className="deployment-support-select-plan__feature"
        key="feature 21"
      >
        <SvgIcon
          name="bullet"
          size="14"
        />
        feature 2
      </li>
    </ul>
    <div
      className="deployment-support-select-plan__price"
    >
      Monthly cost
      <span
        className="deployment-support-select-plan__price-number"
      >
        $
        3600.00
      </span>
    </div>
  </div>
  <div
    className="deployment-support-select-plan__hour-price"
  >
    $
    0.5
     per machine-hour
  </div>
</div>
`;
